src/runtime_src/core/common/info_aie.cpp

```diff
@@ -531,6 +531,52 @@ populate_aie_core_gmio(const boost::property_tree::ptree& pt, boost::property_tr
   pt_array.add_child("gmios",gmio_array);
 }
 
+// This funtion check the duplicate entry in tile_aaray for the same core [row:col]
+bool
+is_duplicate_core(const boost::property_tree::ptree& tile_array, int row, int col)
+{
+  for (auto& tile : tile_array) {
+      if ((tile.second.get<int>("column") == col)
+	  && (tile.second.get<int>("row") == row)) {
+        return true;
+    }
+  }
+
+  return false;
+}
+
+// This funtion populate a specific AIE core which is unused but memory buffers exists [row:col]
+void
+populate_buffer_only_cores(const boost::property_tree::ptree& pt,
+			  const boost::property_tree::ptree& core_info, int gr_id,
+			  boost::property_tree::ptree& tile_array)
+{
+  boost::property_tree::ptree empty_pt;
+
+  for (const auto& g_node : pt.get_child("aie_metadata.EventGraphs", empty_pt)) {
+    if (gr_id != g_node.second.get<int>("id"))
+      continue;
+
+    boost::property_tree::ptree igraph;
+    auto dma_row_it = g_node.second.get_child("dma_rows").begin();
+    for (const auto& node : g_node.second.get_child("dma_columns", empty_pt)) {
+      boost::property_tree::ptree tile;
+      tile.put("column", node.second.data());
+      tile.put("row", dma_row_it->second.data());
+      int row = tile.get<int>("row");
+      int col = tile.get<int>("column");
+
+      // Check whether this core is already added
+      if (is_duplicate_core(tile_array, row, col))
+        continue;
+
+      populate_aie_core(core_info, tile, row, col);
```

Comment:
While not part of this PR, you might wish to reconsider refactoring this function for redundant data is being used (e.g., `tile `contains the `row `and `col `values).

In fact, it looks like the logic to determine if the `tile `is already part of `core_info `should be part of `populate_aie_core()`.

Reply:
Fixed. Thanks